### PR TITLE
gpl update log messages and README for routability driven mode

### PR DESCRIPTION
**Pull request by gudeh**
Improve log descriptions to include detailed information about the stages of routability-driven mode.
Align log output values for better readability.
Add a brief explanation of the routability-driven mode in the readme file.

### CODE REVIEW

**Comment by gudeh** [comment]
This modifications should help to address the requests made in issue #4974.

**Comment by rovinski** [comment]
As long as you're changing this, can I make a request to print areas in um^2 instead of DBU^2? Perhaps with 2 decimal places. The DBU number is almost never useful to anyone.

**Comment by stefanottili** [comment]
[INFO GPL-0012] DieAreaLxLy:    (  0.00,  0.00)
[INFO GPL-0013] DieAreaUxUy:    ( 36.29, 35.84)
[INFO GPL-0014] CoreAreaLxLy:   (  1.14,  1.12)
[INFO GPL-0015] CoreAreaUxUy:   ( 35.15, 34.72)
[INFO GPL-0016] CoreArea:               1142.74

1) LxLy ? UxUy ?

I find lb for left bottom and rt for right top much more meaningful.

2) less is more, why not 

[INFO GPL-0012] DieArea:   ( 0.00 0.00 )  ( 36.29 35.84 ) = …
[INFO GPL-0014] CoreArea:   ( 1.14 1.12 )  ( 35.15 34.72 ) = 1142.74

3) a space between the number and brackets, no need for a comma.
This would make it easier to read for scripting, grep, awk …

**Comment by gudeh** [comment]
> [INFO GPL-0012] DieAreaLxLy:    (  0.00,  0.00)
> [INFO GPL-0013] DieAreaUxUy:    ( 36.29, 35.84)
> [INFO GPL-0014] CoreAreaLxLy:   (  1.14,  1.12)
> [INFO GPL-0015] CoreAreaUxUy:   ( 35.15, 34.72)
> [INFO GPL-0016] CoreArea:               1142.74
> 
> 1) LxLy ? UxUy ?
> 
> I find lb for left bottom and rt for right top much more meaningful.
> 
> 2) less is more, why not 
> 
> [INFO GPL-0012] DieArea:   ( 0.00 0.00 )  ( 36.29 35.84 ) = …
> [INFO GPL-0014] CoreArea:   ( 1.14 1.12 )  ( 35.15 34.72 ) = 1142.74
> 
> 3) a space between the number and brackets, no need for a comma.
> This would make it easier to read for scripting, grep, awk …

Thanks for the suggestions! I will apply them.

File: src/gpl/test/ar01.ok [hunk 1]
```diff
@@ -4,36 +4,36 @@
...
+[INFO GPL-0031] FillerInit:NumGCells:       358
+[INFO GPL-0032] FillerInit:NumGNets:        364
+[INFO GPL-0033] FillerInit:NumGPins:       1122
+[INFO GPL-0023] TargetDensity:             0.60
+[INFO GPL-0024] AvrgPlaceInstArea:         1.94
+[INFO GPL-0025] IdealBinArea:              3.23
+[INFO GPL-0026] IdealBinCnt:                353
+[INFO GPL-0027] TotalBinArea:           1141.67
+[INFO GPL-0028] BinCnt:     16    16
+[INFO GPL-0029] BinSize:  1.76  2.54
```
Comment (rovinski) [inline src/gpl/test/ar01.ok:35]: Coordinate format `(X.XXX Y.YYY)`

File: src/gpl/test/ar01.ok [hunk 2]
```diff
@@ -4,36 +4,36 @@
...
-[INFO GPL-0033] FillerInit: NumGPins: 1122
-[INFO GPL-0023] TargetDensity: 0.60
-[INFO GPL-0024] AveragePlaceInstArea: 7752000
-[INFO GPL-0025] IdealBinArea: 12919999
-[INFO GPL-0026] IdealBinCnt: 353
-[INFO GPL-0027] TotalBinArea: 4566688000
-[INFO GPL-0028] BinCnt: 16 16
-[INFO GPL-0029] BinSize: 3515 5075
-[INFO GPL-0030] NumBins: 256
+[INFO GPL-0003] SiteSize:       (  0.19,  1.40)
```
[the remark this answers is not in the document]
Comment (rovinski) [reply src/gpl/test/ar01.ok:7]: Also please add spaces to both sides of the values: `( 0.19, 1.40 )`

File: src/gpl/test/ar01.ok [hunk 3]
```diff
@@ -4,36 +4,36 @@
...
+[INFO GPL-0010] NumNets:                    364
+[INFO GPL-0011] NumPins:                   1122
+[INFO GPL-0012] DieAreaLxLy:    (  0.00,  0.00)
+[INFO GPL-0013] DieAreaUxUy:    ( 30.40, 42.84)
+[INFO GPL-0014] CoreAreaLxLy:   (  1.14,  1.12)
+[INFO GPL-0015] CoreAreaUxUy:   ( 29.26, 41.72)
+[INFO GPL-0016] CoreArea:               1141.67
+[INFO GPL-0017] NonPlaceInstsArea:         0.00
+[INFO GPL-0018] PlaceInstsArea:          569.77
+[INFO GPL-0019] Util(%):                  49.91
```
Comment (rovinski) [inline src/gpl/test/ar01.ok:23]: Is it possible to add in the units after so it's abundantly clear? e.g.
```
[INFO GPL-0015] CoreBBox:     ( 29.26, 41.72 ) um
[INFO GPL-0016] CoreArea:              1141.67 um^2
[INFO GPL-0019] Util:                    49.91 %
```

**Review by rovinski — COMMENTED** [review]
Thanks for fixing the formatting. A few comments

File: src/gpl/src/placerBase.cpp [hunk 4]
```diff
@@ -1260,33 +1276,78 @@ void PlacerBase::reset()
...
 
-  log_->info(GPL, 12, "DieAreaLxLy: {} {}", die_.dieLx(), die_.dieLy());
-  log_->info(GPL, 13, "DieAreaUxUy: {} {}", die_.dieUx(), die_.dieUy());
-  log_->info(GPL, 14, "CoreAreaLxLy: {} {}", die_.coreLx(), die_.coreLy());
-  log_->info(GPL, 15, "CoreAreaUxUy: {} {}", die_.coreUx(), die_.coreUy());
+  log_->info(GPL,
+             12,
+             "{:16}({:6.2f},{:6.2f})",
+             "DieAreaLxLy:",
+             static_cast<float>(die_.dieLx()) / dbu_per_micron,
```
[the remark this answers is not in the document]
Comment (gudeh) [reply src/gpl/src/placerBase.cpp:1297]: PR #5056 was created to fix this review request.

**Review by maliberty — APPROVED** [review]

**Pull request merged**